Sources/Graphaello/Processing/Pipeline/Component/2. Parsing/SubParsers/SubParser+functionCall+Argument.swift

```diff
@@ -13,11 +13,11 @@ extension SubParser {
     
     static func functionCall(parser: @escaping () -> SubParser<ExprSyntaxProtocol, Argument.QueryArgument>) -> SubParser<FunctionCallExprSyntax, Argument> {
         return .init { expression in
-            guard let calledMember = expression.calledExpression.asProtocol(ExprSyntaxProtocol.self) as? MemberAccessExprSyntax else {
+            guard let calledMember = expression.calledExpression.withoutErasure() as? MemberAccessExprSyntax else {
                 throw ParseError.cannotInstantiateObjectFromExpression(expression, type: Argument.self)
             }
 
-            let argument = Array(expression.argumentList).single()?.expression
+            let argument = Array(expression.argumentList).single()?.expression.withoutErasure()
```

Comment:
You should not need to call `withoutErasure()` here. Without looking too much into the details here, I would suggest that the `parse(from:)` function on `SubParser` takes a `Syntax` or `ExprSyntax` (I don't know the exact context) as an argument. Inside `parse(from:)`, you can then switch on the exact type of the node by
- using `as(SyntaxEnum.self)` (or `as(Syntax.self).as(SyntaxEnum.self)` if `parse(from:)` takes an `ExprSyntax`
- using `as(SyntaxProtocol.self)` or `as(ExprSyntaxProtocol.self)`

Reply:
I love it!